lib/Cron/Workers/SendingQueue/SendingQueue.php

```diff
@@ -111,6 +123,18 @@ public function process($timer = false) {
       $this->mailerTask->configureMailer($newsletter);
       // get newsletter segments
       $newsletterSegmentsIds = $this->newsletterTask->getNewsletterSegments($newsletter);
+      // Pause task in case some of related segments was deleted or trashed
+      if ($newsletterSegmentsIds && !$this->checkDeletedSegments($newsletterSegmentsIds)) {
+        $this->loggerFactory->getLogger(LoggerFactory::TOPIC_NEWSLETTERS)->addInfo(
+          'pause task in sending queue due deleted or trashed segment',
+          ['task_id' => $queue->taskId]
+        );
+        $queue->status = ScheduledTaskEntity::STATUS_PAUSED;
```

Comment:
This pauses the sending queue. Shall we also change the status of the scheduled task?

Reply:
Actually this pauses the scheduled task. The $queue variable in this case refers to [MailPoet\Tasks\Sending](https://github.com/mailpoet/mailpoet/blob/master/lib/Tasks/Sending.php) that combines those 2 models.